### PR TITLE
HSEARCH-3531 Upgrade to Lucene 8.0.0

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net//browse/HSEARCH-3531

### CODE REVIEW

File: backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/extraction/impl/LuceneCollectorsBuilder.java [hunk 1]
```diff
@@ -52,33 +57,58 @@ public DistanceCollector addDistanceCollector(String absoluteFieldPath, GeoPoint
...
+					// Since https://issues.apache.org/jira/browse/LUCENE-8412 (Lucene 8.0.0),
+					// TopFieldCollector returns TopDocs whose ScoreDocs do not contain a score...
+					// Thus we will have to get the scores ourselves.
+
+					// First possibility: if there's a SCORE sort field, use the value of that field
+					SortField[] sortFields = sort.getSort();
+					for ( int i = 0; i < sortFields.length; i++ ) {
+						SortField sortField = sortFields[i];
+						if ( sortField.getType() == SortField.Type.SCORE ) {
+							scoreDocPostProcessor = new LuceneTopFieldCollectorScoreSortPostProcessor( i );
```
Comment (fax4ever) [inline backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/extraction/impl/LuceneCollectorsBuilder.java]: I think that this assignment will be overridden by the other at line 90.

Comment (yrodiere) [reply backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/extraction/impl/LuceneCollectorsBuilder.java]: True; I'll fix that.

**Comment by yrodiere** [comment]
I fixed the problems you mentioned and rebased. Waiting for CI...

**Comment by yrodiere** [comment]
Merged, thanks!

**Pull request merged**